tools/mason/MasonArgParse.chpl

```diff
@@ -186,46 +202,99 @@ module MasonArgParse {
       // make sure all options defined got values if needed
       _checkSatisfiedOptions();
 
+      // assign and missing values their defaults, if supplied
+      _assignDefaultsToMissingOpts();
+
       // check for when arguments passed but none defined
       if argsList.size > 0 && this._actions.size == 0 {
         throw new ArgumentError("unrecognized options/values encountered: " +
                                 " ".join(argsList.these()));
       }
     }
 
+    proc _assignDefaultsToMissingOpts() {
+      // set any default values as needed
+      for name in this._actions.keys() {
+        const act = this._actions.getBorrowed(name);
+        const arg = this._result.getReference(name);
+        if !arg._present && !act._defaultValue.isEmpty() {
```

Comment:
Is it possible somebody would want a default value to be the empty list? It seems possible to me. If that's possible then we need a `var _hasDefault` field in `class Action`. There are many ways to set this according to whether or not `defaultValue` was provided but my favorite is probably to use `defaultValue:?t = none`. That makes it generic with a default value of `none`. Then you would check in the body if `isNoneType(defaultValue)` which means no default was provided.

Reply:
I'm not clear on the scenario where defining an empty list as default would be more useful than just getting `.values()` back from the argument, which would be empty if nothing was entered from the command line. However, due in part to implementing the suggestion for `defaultValue` to be generic, both here and below, I will add the `_hasDefault` flag to `Action`